src/go/rpk/pkg/cli/cmd/acl/common.go

```diff
@@ -57,26 +58,90 @@ type (
 	}
+
 	aclWithMessage struct {
-		Principal           string
-		Host                string
-		ResourceType        kmsg.ACLResourceType
-		ResourceName        string
-		ResourcePatternType kmsg.ACLResourcePatternType
-		Operation           kmsg.ACLOperation
-		Permission          kmsg.ACLPermissionType
-		Message             string
+		Principal           string                      `json:"principal" yaml:"principal"`
+		Host                string                      `json:"host" yaml:"host"`
+		ResourceType        kmsg.ACLResourceType        `json:"resource_type" yaml:"resource_type"`
+		ResourceName        string                      `json:"resource_name" yaml:"resource_name"`
+		ResourcePatternType kmsg.ACLResourcePatternType `json:"resource_pattern_type" yaml:"resource_pattern_type"`
+		Operation           kmsg.ACLOperation           `json:"operation" yaml:"operation"`
+		Permission          kmsg.ACLPermissionType      `json:"permission" yaml:"permission"`
+		Message             string                      `json:"message" yaml:"message"`
+	}
+
+	// Used for json/yaml formatted output in "rpk acl lists|delete".
+	describedACLsResult struct {
+		Principal *string `json:"principal" yaml:"principal"` // Principal is the optional user that was used in this filter.
+		Host      *string `json:"host" yaml:"host"`           // Host is the optional host that was used in this filter.
+
+		Type       kmsg.ACLResourceType   `json:"type" yaml:"type"`             // Type is the type of resource used for this filter.
+		Name       *string                `json:"name" yaml:"name"`             // Name is the name of the resource used for this filter.
+		Pattern    kadm.ACLPattern        `json:"pattern" yaml:"pattern"`       // Pattern is the name pattern used for this filter.
+		Operation  kmsg.ACLOperation      `json:"operation" yaml:"operation"`   // Operation is the operation used for this filter.
+		Permission kmsg.ACLPermissionType `json:"permission" yaml:"permission"` // Permission is permission used for this filter.
+
+		Err error `json:"error" yaml:"error"` // Err is non-nil if this filter has an error.
+	}
+
+	filteredAndDescribed struct {
+		Filter describedACLsResult `json:"filter" yaml:"filter"`
+		ACLS   []acl               `json:"acls" yaml:"acls"`
 	}
 )
 
+// filteredAndDescribedResults and methods.
+type filteredAndDescribedResults struct {
+	Results []filteredAndDescribed `json:"filtered_acls" yaml:"filtered_acls"`
+}
+
+func (collection *filteredAndDescribedResults) addFilterAndDescribed(filteredACL filteredAndDescribed) {
+	collection.Results = append(collection.Results, filteredACL)
+}
+
+// =======================
+
+// deletedACLCollection and methods.
+type deletedACLCollection struct {
+	Deleted []filteredAndDescribed `json:"deleted_acls" yaml:"deleted_acls"`
+}
+
+func (collection *deletedACLCollection) AddACL(deletedACL filteredAndDescribed) {
+	collection.Deleted = append(collection.Deleted, deletedACL)
+}
```

Comment:
I'm not sure of the purpose of this struct.

I'd prefer an append at the callsite (I might make this comment a few times since I'm jumping around the PR), so AddACL can be dropped.

w.r.t. encoding, is there a reason to favor encoding a struct with `deleted_acls` being an array, vs. just encoding the array?

Reply:
Will move the various Add* helper methods to the call location as suggested.  For the struct, the idea I was going for was that the return value from the command should always be a key => value pair in the final json/yaml output for consistency.

So in the event of the output of the command only being an array, it should at least be thing => array. I'm happy to change that if the output from the rpk command line isn't going to be treated as an "api like thing" but I think that's probably a good idea idea in the long run?

As far as putting it in a struct, that was because I thought that's the only way to be be able to add tags for json/yaml printing. If it's not, then I'm happy to research the language more to not create structs that are just containers with a single member for json/yaml printing, it does feel very verbose.